src/labeler.py

```diff
@@ -77,3 +77,39 @@ def priority_queues(all_problems):
                     label = 1
         all_labels.append(label)
     return all_labels
+
+
+def loops(all_problems):
+    all_labels = []
+    for problem in all_problems:
+        label = 0
+        for solution in problem['solutions']:
+            for line in solution.split('\n'):
+                if ' for ' in line.lower() or ' do ' in line.lower() or ' while ' in line.lower():
```

Comment:
I like this simple solution of searching for a space after every keyword. However, also searching for a space before the word would miss the match if the word occurs at the very beginning of a line, so there's no space before it.  this applies to the other functions too. I realize why you did it to avoid things like `String word_for = "..."` but to to catch everything you might need another technique. 

I recommend using regex. In case you aren't familiar with it and would like help, feel free to msg me about it. I think Isabel also has experience using it.